### PR TITLE
Misc fixes

### PR DESCRIPTION
**Pull request by jschueller**
- #1395
- #1456
- #1448
- #1461
- #1471 
- ProcessSample __getitem__/__setitem__ to deal with Sample instead of Field

### CODE REVIEW

**Comment by jschueller** [comment]
@sofianehaddad I made the small adjustement in EGO

**Comment by regislebrun** [comment]
+1

File: ChangeLog [hunk 1]
```diff
@@ -23,6 +23,7 @@
 
 === Python module ===
  * ProcessSample __getitem__ returns Sample instead of Field
+ * Implement list indexing
```
Comment (sofianehaddad) [inline ChangeLog:26]: This fixes also https://github.com/openturns/openturns/issues/1395

File: lib/src/Base/Solver/Fehlberg.cxx [hunk 2]
```diff
@@ -244,7 +245,7 @@ Sample Fehlberg::solve(const Point & initialState,
 /* Perform one step of the Fehlberg method
    See J. Stoer, R. Bulirsch, "Introduction to Numerical Analysis 2nd Edition", pp448-458.
  */
-Point Fehlberg::computeStep(Function & transitionFunction,
+Point Fehlberg::computeStep(Pointer<EvaluationImplementation> & transitionFunction,
```
Comment (regislebrun) [inline lib/src/Base/Solver/Fehlberg.cxx:248]: THE trick!

**Pull request merged**